clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -478,7 +479,8 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         shard->prev_instr_ = memref;
         if (shard->prev_instr_decoded_ != nullptr)
             instr_free(GLOBAL_DCONTEXT, shard->prev_instr_decoded_.get());
```

Comment:
We are still missing an instr_free at the end for prev_instr_decoded_, which is what the leak was. Does this PR pass the asan check? I suspect it might not.

To fix this, we should make the prev_instr_decoded_ also a unique_ptr<instr_autoclean_t> so that prev_instr_decoded_ is cleaned-up automatically.

Reply:
Done.